### PR TITLE
package-set-0.12

### PR DESCRIPTION
**Pull request by chexxor**
This adds the SHAs for the core packages (those in the purescript org), from this list: https://github.com/purescript/purescript/issues/3116

I need to figure out how to verify this package.json file - it looks like `psc-package verify` expects a psc-package.json file...

Until then, though, I welcome reviews.

### CODE REVIEW

**Comment by chexxor** [comment]
Will probably need to remove all non-core packages.

**Comment by justinwoo** [comment]
Should add this as a branch first? like 0.12-rc or something.

**Comment by LiamGoodacre** [comment]
Yeah you'll need to create a new branch in this repo and then update this PR to switch base branch.

**Comment by justinwoo** [comment]
Made one: https://github.com/purescript/package-sets/tree/0.12-rc

**Comment by LiamGoodacre** [comment]
Will eventually need to update travis.yml to point at the 0.12-rc compiler release once that is available.

**Comment by justinwoo** [comment]
Could you reset the commit and force push it? Travis seems to not understand how to use the correct commit.

File: packages.json [hunk 1]
```diff
@@ -1257,21 +1235,12 @@
     "repo": "https://github.com/Thimoteus/purescript-mmorph.git",
     "version": "v3.0.0"
   },
-  "monoid": {
```
[the remark this answers is not in the document]
Comment (LiamGoodacre) [reply packages.json]: Monoid is deprecated.

[the remark this answers is not in the document]
Comment (LiamGoodacre) [reply packages.json]: The type class has moved to Prelude.

[the remark this answers is not in the document]
Comment (justinwoo) [reply packages.json]: Ah, then all the listings of dependencies should be removed. Cool

[the remark this answers is not in the document]
Comment (LiamGoodacre) [reply packages.json]: Symbols too https://github.com/purescript/purescript-symbols/tree/compiler/0.12

**Comment by justinwoo** [comment]
In addition, reverse dependencies of typelevel-prelude won't work without upgrades themselves, so they need to be removed. E.g. Simple-JSON

**Comment by chexxor** [comment]
Yeah, I'll work on that.

**Comment by chexxor** [comment]
I just removed all non-core libs. Can re-add other libs as they get upgraded to work with new core libs.

**Comment by chexxor** [comment]
Requires some PRs be merged in some libs:
https://github.com/purescript/purescript-exceptions/pull/28
https://github.com/purescript/purescript-inject/pull/10 (also require some edits before merging)

**Comment by chexxor** [comment]
Removed "inject" package, as that'll be merged into functors/either. https://github.com/purescript/purescript-inject/pull/10

**Comment by justinwoo** [comment]
Exceptions needs an update or a new target?

**Comment by chexxor** [comment]
Yeah I suspected of was a caching issue in CI, which seems hard to debug so
I gave up to look again at a later time when I have more fortitude.

**Pull request merged**